cmd/queue/main.go

```diff
@@ -298,8 +257,19 @@ func main() {
 		logger.Fatal("Error creating new config", zap.Error(err))
 	}
 	kubeClient = kc
-	go connectStatSink()
+
+	go func() {
+		autoscalerEndpoint := fmt.Sprintf("ws://%s.%s.svc.cluster.local:%s", servingAutoscaler, system.Namespace, servingAutoscalerPort)
+		logger.Infof("Connecting to autoscaler at %s", autoscalerEndpoint)
+		conn, err := websocket.NewDurableSendingConnection(autoscalerEndpoint)
+		if err != nil {
+			logger.Fatal("Failed to connect to autoscaler, shutting down", zap.Error(err))
```

Comment:
This is a change in behavior.  Previously if a pod was unable to connect to the autoscaler, it would keep retrying indefinitely.  Now it will terminate after the maximum number of attempts.

So what is the right thing to do in the case a pod cannot connect to the autoscaler?  I think that we should do nothing.  That is, the autoscaler should emit a warning that it's not getting any metrics.  And the pod should emit a warning that it can't send any metrics.  But we should not stop serving.

Reply:
Yep, fine with me. Will change that.

Reply:
Perhaps the exponential backoff function should have an option for "never give up" and it will keep retrying at a constant rate once it reaches its maximum duration.

Or you can just give up and log an error (but not fatal).